avocado/core/parser.py

```diff
@@ -172,3 +178,94 @@ def finish(self):
+        for kwarg in kwargs.split(','):
+            if kwarg == '':
+                continue
+            key, value = kwarg.split('=')
+            result[key] = value
+        return result
+
+    def _get_resolutions_by_kind(self, kind, paths):
+        self.validate_kind_section(kind)
+
+        resolutions = []
+        uri = self._get_uri_from_section(kind)
+        args = self._get_args_from_section(kind)
+        kwargs = self._get_kwargs_from_section(kind)
+        success = ReferenceResolutionResult.SUCCESS
+        if uri == '$test':
+            for path in paths:
+                runnable = Runnable(kind, path, *args, **kwargs)
+                resolutions.append(ReferenceResolution(reference=path,
+                                                       result=success,
+                                                       resolutions=[runnable],
+                                                       origin='hint-file'))
+        elif '$test' in args:
+            for path in paths:
+                runnable = Runnable(kind, uri, path, **kwargs)
+                resolutions.append(ReferenceResolution(reference=path,
+                                                       result=success,
+                                                       resolutions=[runnable],
+                                                       origin='hint-file'))
+        return resolutions
+
+    def _get_uri_from_section(self, section):
+        return self.config.get(section, 'uri')
+
+    def _parse(self):
+        self.config = ConfigParser()
+        config_paths = self.config.read(self.filename)
+        if not config_paths:
+            raise settings.ConfigFileNotFound(self.filename)
+
+    def get_resolutions(self):
+        """Return a list of resolutions based on the file definitions."""
+        resolutions = []
+        for kind in self.config['kinds']:
+            files = self.config.get('kinds', kind)
+            resolutions.extend(self._get_resolutions_by_kind(kind,
+                                                             glob(files)))
+        return resolutions
+
+    def validate_kind_section(self, kind):
+        """Validates a specific "kind section".
+
+        This method will raise a `settings.SettingsError` if any problem is
+        found on the file.
+
+        :param kind: a string with the specific section.
+        """
+        if kind not in self.config:
+            msg = 'Section {} is not defined. Please check your hint file.'
+            raise settings.SettingsError(msg.format(kind))
+
+        uri = self._get_uri_from_section(kind)
+        args = self._get_args_from_section(kind)
+        kwargs = self._get_kwargs_from_section(kind)
+        if uri is None:
+            msg = "uri needs to be defined inside {}".format(kind)
+            raise settings.SettingsError(msg)
+
+        count = [uri, *args, kwargs].count('$test')
+        if count != 1:
```

Comment:
I tend to stay away from policies such as this. There *may* be a case where an user would want to have both `uri` and `args` set to `$test` (or `$path` as I suggested previously). Can you think of a good to actively prevent it from being possible?

Reply:
@clebergnu in fact, I could not think on `kwargs`, maybe make this `$testpath` available only for `uri` and `args` ? What do you think ?

Reply:
If it doesn't add (much) to the code complexity, I'd make the variable expansion a generic feature, for any key. There may be "weird" test cases out there that will take the path as a environment variable, for example.

Reply:
I got your point now. Ok, will do this.